src/ops/array.rs

```diff
@@ -94,32 +98,35 @@ pub(crate) fn add_by_internal(value: &mut [u32], by: &[u32]) -> u32 {
 }
 
 pub(crate) fn add_by_internal_flattened(value: &mut [u32; 3], by: u32) -> u32 {
-    let mut carry: u64;
-    let mut sum: u64;
-    sum = u64::from(value[0]) + u64::from(by);
-    value[0] = (sum & U32_MASK) as u32;
-    carry = sum >> 32;
-    if carry > 0 {
-        sum = u64::from(value[1]) + carry;
-        value[1] = (sum & U32_MASK) as u32;
-        carry = sum >> 32;
-        if carry > 0 {
-            sum = u64::from(value[2]) + carry;
-            value[2] = (sum & U32_MASK) as u32;
-            carry = sum >> 32;
-        }
-    }
-    carry as u32
+    manage_add_by_internal(by, value)
 }
 
 #[inline]
-pub(crate) fn add_one_internal(value: &mut [u32]) -> u32 {
-    let mut carry: u64 = 1; // Start with one, since adding one
-    let mut sum: u64;
-    for i in value.iter_mut() {
-        sum = (*i as u64) + carry;
-        *i = (sum & U32_MASK) as u32;
-        carry = sum >> 32;
+pub(crate) fn add_one_internal(value: &mut [u32; 3]) -> u32 {
+    manage_add_by_internal(1, value)
+}
+
+// `u64 as u32` are safe because of widening and 32bits shifts
+#[inline]
+pub(crate) fn manage_add_by_internal<const N: usize>(initial_carry: u32, value: &mut [u32; N]) -> u32 {
```

Comment:
Used by `add_one_internal` and `add_by_internal_flattened` as both are almost the same. Other similar functions could also leverage this common entry point to remove duplication.

Reply:
Yes, I think after we optimized `from_str_radix_n` these functions could be removed.